py/picca/pk1d/postproc_pk1d.py

```diff
@@ -862,12 +818,370 @@ def compute_average_pk_redshift(
+            for iboot in range(number_bootstrap):
+                if bootid[iboot] is None:
+                    p1d_weights_z, covariance_weights_z, p1d_groups_z = [], [], []
+                else:
+                    p1d_weights_z = p1d_groups[izbin][1][bootid[iboot]]
+                    covariance_weights_z = p1d_groups[izbin][2][bootid[iboot]]
+                    p1d_groups_z = p1d_groups[izbin][3][bootid[iboot]]
+                p1d_groups_bootstrap.append(
+                    [
+                        izbin,
+                        p1d_weights_z,
+                        covariance_weights_z,
+                        p1d_groups_z,
+                    ]
+                )
+
+        func = partial(
+            compute_cov,
+            weight_method,
+            mean_p1d_table,
+            nbins_k,
+        )
+        if number_worker == 1:
+            output_cov = [func(*p) for p in p1d_groups_bootstrap]
+        else:
+            with Pool(number_worker) as pool:
+                output_cov = pool.starmap(func, p1d_groups_bootstrap)
+
+        for izbin in range(nbins_z):
+            boot_cov = []
+            for iboot in range(number_bootstrap):
+                covariance_array = (output_cov[izbin * number_bootstrap + iboot],)
+                boot_cov.append(covariance_array)
+
+            i_min = izbin * nbins_k * nbins_k
+            i_max = (izbin + 1) * nbins_k * nbins_k
+            cov_table["boot_covariance"][i_min:i_max] = np.mean(boot_cov, axis=0)
+            cov_table["error_boot_covariance"][i_min:i_max] = np.std(boot_cov, axis=0)
+
+
+
+
+def compute_p1d_groups(
+    weight_method,
+    nbins_k,
+    zbin_edges,
+    izbin,
+    p1d_table,
+    k_index,
+    snrfit_table,
+    number_worker,
+):
+    """Compute the P1D groups before covariance matrix calculation.
+    Put all the P1D in the same k grid.
+
+    Arguments
+    ---------
+    weight_method: str,
+    Method to weight the data.
+
+    nbins_k (int):
+    Number of k bins.
+
+    zbin_edges (array-like):
+    All redshift bins.
+
+    izbin (int):
+    Current redshift bin being considered.
+
+    p1d_table (array-like):
+    Table of 1D power spectra, with columns 'Pk' and 'sub_forest_id'.
+
+    k_index (array-like):
+    Array of indices for k-values, with -1 indicating values outside of the k bins.
+
+    snrfit_table: numpy ndarray,
+    Table containing SNR fit infos
+
+    number_worker: int
+    Number of workers for the parallelization
+
+    Return
+    ------
+    p1d_weights  (array-like):
+    Weights associated with p1d pixels for all subforest, used in the calculation of covariance.
+
+    covariance_weights (array-like):
+    Weights for all subforest used inside the main covariance sum.
+
+    p1d_groups (array-like):
+    Individual p1d pixels grouped in the same wavenumber binning for all subforest
+    """
+
+    select_z = (p1d_table["forest_z"] < zbin_edges[izbin + 1]) & (
+        p1d_table["forest_z"] > zbin_edges[izbin]
```

Comment:
this is the `p1d_table` of each chunk and not the `mean_p1d_table` of the combined stats, so my comments regarding indexing shouldn't apply here...